flyway-core/src/main/java/org/flywaydb/core/internal/dbsupport/DryRun.java

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright 2010-2016 Boxfuse GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.flywaydb.core.internal.dbsupport;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.sql.Savepoint;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.WeakHashMap;
+import org.flywaydb.core.api.FlywayException;
+
+public class DryRun {
+    private static final String SAVEPOINT_MARKER = "dryrun";
+    private final Map<Connection, Savepoint> dryRunSavepoints = new WeakHashMap<Connection, Savepoint>();
+    private final int connectionCount;
+    
+    public DryRun(int transactionIsolationLevel, DbSupport dbSupport, Connection... connections) {
+        if (dbSupport == null) {
+            throw new NullPointerException("A Dry-run requires dbSupport to determine whether connections can support a rollback successfully.");
+        }
+
+        if (!dbSupport.supportsDdlTransactions()) {
```

Comment:
Truncate IS, however, DDL Transaction safe, just not MVCC safe.  Considering we're not executing concurrent transactions for doing a dry-run of a migration, I don't see how that's a problem.  If there was a concurrent transaction (like we're doing a dry-run on an existing database that's connected to a running applicaiton) that would be a side effect which should be noted in any documentation.  But since nothing can really be done about it other than "Don't do that" - I'm not sure I see a big issue with the implementation.

Do you have an idea how to get around such issues?  This feature is basically what's preventing my company (and many others) from using a neat tool like Flyway in production.  Assuming the known caveats are acceptable, I wouldn't want to just not implement a feature because it's not perfect in all situations.